File: problem1-5/problem2/problem2.hs
-- By considering the terms in the Fibonacci sequence whose
-- values do not exceed four million, find the sum of the
-- even-valued terms.
soln = sum [x | x <- takeWhile (<= 4000000) fib, even x]
  where
    fib = 1 : 1 : zipWith (+) fib (tail fib)
ans = show soln

main = do
  putStrLn $ "Solution: " ++ ans

File: problem1-5/problem2/problem2.rs
fn main() {
    let mut ans = 0;
    let mut fib_one = 1;
    let mut fib_two = 2;
    let mut fib_tres;
    while (fib_two < 4000000) {
        if (fib_two % 2 == 0) {
            ans += fib_two
        }
        fib_tres = fib_two + fib_one;
        fib_one = fib_two;
        fib_two = fib_tres;
    };
    println(fmt!("Solution %?", ans));
}
